### PR TITLE
Jobs now queued by job type.

### PR DESCRIPTION
**Pull request by kd7uiy**
Continuing to work on a new job system, this new version now supports character priorities. Distance is not yet factored in.

### CODE REVIEW

File: Assets/Scripts/Models/Job/JobManager.cs [hunk 1]
```diff
@@ -71,38 +79,70 @@ public Job GetJob(Character character)
...
     }
 
     public void Remove(Job job)
     {
-        jobQueue.Remove(job);
+        jobQueue[job.Category].Remove(job);
+
+        if (OnJobRemoved != null)
+        {
+            OnJobCreated(job);
```
Comment (koosemose) [inline Assets/Scripts/Models/Job/JobManager.cs]: Wrong event called.

File: Assets/Scripts/Models/Job/JobManager.cs [hunk 2]
```diff
@@ -140,4 +190,10 @@ private void DebugLog(string message, params object[] par)
     {
         UnityDebugger.Debugger.LogFormat("FSM", message, par);
     }
+
+    [System.Diagnostics.Conditional("FSM_DEBUG_LOG")]
+    private void DebugLogError(string message, params object[] par)
+    {
+        UnityDebugger.Debugger.LogErrorFormat("FSM", message, par);
+    }
```
Comment (koosemose) [inline Assets/Scripts/Models/Job/JobManager.cs]: I don't think these sorts of functions should be needed anymore, they were only a stopgap before implementing our logger and probably should have been removed once that was done, almost certainly don't need to add any more.

File: Assets/Scripts/Models/Job/JobManager.cs [hunk 3]
```diff
@@ -13,26 +13,29 @@
...
 
-    // Returns the job count in the queue.
-    // (Necessary, since jobQueue is private).
-    public int GetCount()
-    {
-        return jobQueue.Count;
-    }
+    public event JobChanged OnJobModified;
+
+    public event JobChanged OnJobRemoved;
```
Comment (koosemose) [inline Assets/Scripts/Models/Job/JobManager.cs]: These should be in the format of JobChanged, without the "On". The "On<action>" format is what we would use for the functions that get passed to these (the observer I suppose).

File: Assets/Scripts/Models/Job/JobManager.cs (hunk 2 above)
Comment (NogginBops) [reply Assets/Scripts/Models/Job/JobManager.cs]: This also creates a problem where clicking on the error in the editor doesn't get you to the line where the error was actually printed.

Comment (kd7uiy) [reply Assets/Scripts/Models/Job/JobManager.cs]: Good point, I made a better change in my newest PR.

**Pull request merged**